### ISSUE TITLE
Watch taking a long time to load configuration

### ISSUE THREAD

**Issue by thenoelforte** [gruntjs/grunt-contrib-watch#414]
Someone else may have already spotted this issue, but in my environment, when the Gruntfile is updated, the command line will output:

    Running "watch" task

as opposed to:

    Running "watch task
    Waiting...

Running `grunt --verbose` shows that on first run, the watch task executes successfully and all directories and files are being watched, however on saving the Gruntfile the command line outputs the following:

    Reloading watch config...

    Reading "Gruntfile.js" Gruntfile...OK

    Registering Gruntfile tasks.
    Reading package.json...OK
    Parsing package.json...OK
    Initializing config...OK

...followed by task loading and other stuff, eventually getting to...

    Loading "Gruntfile.js" tasks...OK
    + build, default, serve

    Running "watch" task

But no files are being watched. Thus, I have to restart grunt in order for the watch task to begin picking up my files again.

Running on `node 0.12.0`, `grunt-cli 0.1.13`, `grunt 0.4.5`

**Comment by thenoelforte** [comment]
Upate on this: I did some testing, and it turns out that the presence of `grunt-newer` and `grunt-contrib-imagemin` drastically increase the number of files that `grunt-contrib-watch` has to watch. I tried excluding the `node_modules` directory using the following in file match patterns

   ...'!node_modules/**'...

but this only adds lag time from when `Watching...` is returned in the command line to when the files are actually being watched.

If someone could help me resolve this issue that would be great.
